area.go

```diff
@@ -2,50 +2,161 @@ package cursor
 
 import (
 	"fmt"
+	"io"
 	"os"
+	"runtime"
 	"strings"
 )
 
 // Area displays content which can be updated on the fly.
 // You can use this to create live output, charts, dropdowns, etc.
 type Area struct {
-	height int
-	writer Writer
+	height     int
+	writer     io.Writer
+	cursor     *Cursor
+	cursorPosY int
 }
 
 // NewArea returns a new Area.
-func NewArea() Area {
-	return Area{
+func NewArea() *Area {
+	return &Area{
 		writer: os.Stdout,
-		height: 0,
+		cursor: cursor,
 	}
 }
 
-// WithWriter sets a custom writer for the Area.
-func (area Area) WithWriter(writer Writer) Area {
+// WithWriter sets the custom writer
+func (area *Area) WithWriter(writer io.Writer) *Area {
 	area.writer = writer
-
+	area.cursor = area.cursor.WithWriter(writer)
 	return area
 }
 
 // Clear clears the content of the Area.
 func (area *Area) Clear() {
-	Bottom()
-
+	// Initialize writer if not done yet
+	if area.writer == nil {
+		area.writer = os.Stdout
+	}
 	if area.height > 0 {
-		ClearLinesUp(area.height)
+		area.Bottom()
+		area.ClearLinesUp(area.height)
+	} else {
+		area.cursor.ClearLine()
 	}
 }
 
 // Update overwrites the content of the Area.
 func (area *Area) Update(content string) {
-	oldWriter := target
-
-	SetTarget(area.writer) // Temporary set the target to the Area's writer so we can use the cursor functions
 	area.Clear()
-	SetTarget(oldWriter) // Reset the target to the old writer
 	fmt.Fprintln(area.writer, content)
+	// Detect height of cursor area
+	area.height = strings.Count(content, "\n")
+	if strings.HasSuffix(content, "\n") || runtime.GOOS == "windows" {
```

Comment:
Could you please add a comment here, why windows has a different behavior?

Reply:
Changed to not require that special handling at all and not add an unneeded newline (use `Fprint` instead of `Fprintln`). Adjustments might be required for pterm `InteractiveTextInputPrinter` concerning cursor movement (which also has issues with the current version)